executor/runner/runner.go

```diff
@@ -301,43 +301,46 @@ no_launchguard:
 		err = r.maybeSetupExternalLogger(ctx, logDir)
 		if err != nil {
 			r.logger.Error("Unable to setup logging for container: ", err)
-			r.updateStatus(ctx, titusdriver.Lost, err.Error())
+			r.updateStatusWithDetails(ctx, titusdriver.Lost, err.Error(), details)
 			return
 		}
 	} else {
 		r.logger.Info("Not starting external logger")
 	}
 
-	// TODO(fabio): Start should return Details
-	details, err := r.runtime.Details(r.container)
-	if err != nil {
-		r.logger.Error("Error fetching details for task: ", err)
-		r.updateStatus(ctx, titusdriver.Lost, err.Error())
-		return
-	} else if details == nil {
-		r.logger.Error("Unable to fetch task details")
+	if details == nil {
+		r.logger.Fatal("Unable to fetch task details")
 	}
 	r.metrics.Counter("titus.executor.taskLaunched", 1, nil)
-	r.updateStatusWithDetails(ctx, titusdriver.Running, "running", details)
 
-	// report metrics for startup time, docker image size
-	r.metrics.Timer("titus.executor.containerStartTime", time.Since(startTime), r.container.ImageTagForMetrics())
+	r.monitorContainer(ctx, startTime, statusChan, details)
+}
 
-	ticks := time.NewTicker(r.config.StatusCheckFrequency)
-	defer ticks.Stop()
+func (r *Runner) monitorContainer(ctx context.Context, startTime time.Time, statusChan <-chan runtimeTypes.StatusMessage, details *runtimeTypes.Details) {
+	lastMessage := ""
+	runningSent := false
 
 	for {
 		select {
-		case <-ticks.C:
-			status, err := r.runtime.Status(r.container)
-			if err != nil {
-				r.logger.Error("Status result error: ", err)
-			}
-			shouldQuit, titusTaskStatus, msg := parseStatus(status, err)
-			if shouldQuit {
-				r.logger.Info("Status: ", titusTaskStatus.String())
-				// TODO: Generate Update
-				r.updateStatus(ctx, titusTaskStatus, msg)
+		case statusMessage := <-statusChan:
```

Comment:
this needs to handle `statusChan` being closed too, no? It seems it will be closed when we can't talk to the docker daemon anymore, in which case this will cause a `panic: nil pointer`. At least failing more gracefully and sending the `StatusUpdate.Lost` message may be better.

Reply:
:+1: